code/game/objects/gamemode/crash/machines.dm

```diff
@@ -0,0 +1,216 @@
+// -- Print disc computer
+/obj/item/circuitboard/computer/nuke_disc_generator
+	name = "Circuit board (Nuke Disc Generator)"
+	build_path = /obj/machinery/computer/engine_computer
+
+
+/obj/machinery/computer/nuke_disc_generator
+	name = "Nuke Disc Generator"
+	desc = "Used to generate the correct auth discs for the nuke."
+	icon_state = "atmos" // TODO: Better icon state
+	circuit = "/obj/item/circuitboard/computer/nuke_disc_generator"
+
+	resistance_flags = INDESTRUCTIBLE|UNACIDABLE
+
+	var/generate_time = 5 SECONDS // time for the machine to generate the disc
+	var/segment_time = 1 SECONDS // time to start the hack
+	var/total_segments = 5 // total number of times the hack is required
+	var/completed_segments = 0 // what segment we are on, (once this hits total, disc is printed)
+	var/current_timer
+
+	var/disc_type = null
+	var/obj/item/disk/nuclear/crash/disc
+
+/obj/machinery/computer/nuke_disc_generator/Initialize()
+	. = ..()
+
+	if(!disc_type)
+		stack_trace("disc_type is required to be set before init")
```

Comment:
Since only the base type doesn't have it, is this a necessary stack trace?

Reply:
I just wanted to add it so someone knew why the computer was disappearing.

Reply:
This will runtime and the unga dunga admin won't care much, if it's meant to do that on the base type, the stack trace is not really useful.

Reply:
its really for mappers was what i meant.

Reply:
then make it a WARNING

Reply:
Done